python/runtime/xgboost/predict.py

```diff
@@ -78,23 +78,11 @@ def predict_and_store_result(bst, dpred, feature_file_id, model_params,
                              feature_column_names, feature_metas, is_pai, conn,
                              result_table):
     preds = bst.predict(dpred)
-    # TODO(yancey1989): output the original result for various
-    # objective function.
-    if model_params:
-        obj = model_params["objective"]
-        if obj.startswith("binary:"):
-            preds = (preds > 0.5).astype(int)
-        elif obj.startswith("multi:"):
-            preds = np.argmax(np.array(preds), axis=1)
-        else:
-            # using the original prediction result of predict API by default
-            pass
-    else:
-        # prediction output with multi-class job has two dimensions, this
-        # is a temporary way, can remove this else branch when we can load
-        # the model meta not only on PAI submitter.
-        if len(preds.shape) == 2:
-            preds = np.argmax(np.array(preds), axis=1)
+    # prediction output with multi-class job has two dimensions, this
+    # is a temporary way, can remove this else branch when we can load
+    # the model meta not only on PAI submitter.
+    if len(preds.shape) == 2:
+        preds = np.argmax(np.array(preds), axis=1)
```

Comment:
Why change this? I think that `model_params` has been got [here](https://github.com/sql-machine-learning/sqlflow/pull/2876/files#diff-009614f1b6ee464541b3aec9b86ad5d0L60). And we can get `objective` here so that we can distinguish binary classification, multi-class classification, and regression models.

Reply:
`preds` here is already the classification label here, no need to do `argmax` anymore.

Reply:
I see.